cis/libs/streams.py

```diff
@@ -0,0 +1,57 @@
+import base64
+import json
+from cis.settings import get_config
+
+
+class Operation(object):
+    def __init__(self, boto_session, publisher, signature, encrypted_profile_data):
+        self.boto_session = boto_session
+        self.config = get_config()
+        self.encrypted_profile_data = encrypted_profile_data
+        self.publisher = publisher
+        self.signature = signature
+        self.kinesis_client = None
+
+    def to_kinesis(self):
+        """
+        Publish data to CIS kinesis stream given a partition key.
+    
+        :data: Data to be published to kinesis (dict)
+        :partition_key: Kinesis partition key used to publish data to
+        """
+
+        if self.kinesis_client is None:
```

Comment:
The way that you have coded it ``self.kinesis_client`` is always ``None`` when running this method. Please move the initialization to ``__init__``.